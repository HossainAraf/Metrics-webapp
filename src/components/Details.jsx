import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import PropTypes from 'prop-types';

const Card = ({ label, value }) => (
  <li className="">
    <div className="">{label}</div>
    <div className="">{value}</div>
  </li>
);

const Details = () => {
  const params = useParams();
  const { details } = params;
  const countries = useSelector((state) => state.countries.countries);

  const selectedCountry = countries.find((country) => country.name === details);

  return (
    <div>
      <div className="">
        <div className="">
          <div>
            <img src={selectedCountry?.flag} alt={details} className="" />
          </div>
        </div>
        <div className="">
          <h1 className="">{details?.toUpperCase()}</h1>
          <h3 className="">{`Population: ${selectedCountry?.population}`}</h3>
        </div>
      </div>
      <div className="">STATS BY COUNTRY</div>
      <div>
        <ul>
          <Card label="CAPITAL" value={selectedCountry?.capital} />
          <Card label="POPULATION" value={selectedCountry?.population} />
          <Card label="AREA" value={selectedCountry?.area} />
          <Card label="CURRENCY" value={selectedCountry?.currencies[0]?.code} />
          <Card label="CALLING CODE" value={selectedCountry?.callingCodes} />
          <Card label="TIMEZONES" value={selectedCountry?.timezones} />
        </ul>
      </div>
    </div>
  );
};

Card.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
};

export default Details;
