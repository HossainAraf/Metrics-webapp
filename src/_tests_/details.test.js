import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Details from '../components/Details';

describe('Details Component', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        {' '}
        <Details />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
