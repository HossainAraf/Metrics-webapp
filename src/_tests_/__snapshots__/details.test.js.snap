// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component should match the snapshot 1`] = `
Array [
  " ",
  <div>
    <div
      className=""
    >
      <div
        className="top-container"
      >
        <div
          className="img-wrapper"
        >
          <img
            className=""
          />
        </div>
        <div
          className="country-title"
        >
          <li
            className=""
          >
            <div
              className=""
            />
            <div
              className=""
            />
          </li>
          <li
            className=""
          >
            <div
              className=""
            />
            <div
              className=""
            />
          </li>
        </div>
      </div>
    </div>
    <div
      className="center"
    >
      COUNTRY BREAKDOWN - 2023
    </div>
    <div
      className="country-details-wrapper"
    >
      <ul
        className="country-details"
      >
        <li
          className=""
        >
          <div
            className=""
          >
            CAPITAL
          </div>
          <div
            className=""
          />
        </li>
        <li
          className=""
        >
          <div
            className=""
          >
            POPULATION
          </div>
          <div
            className=""
          />
        </li>
        <li
          className=""
        >
          <div
            className=""
          >
            AREA
          </div>
          <div
            className=""
          />
        </li>
        <li
          className=""
        >
          <div
            className=""
          >
            CURRENCY
          </div>
          <div
            className=""
          />
        </li>
        <li
          className=""
        >
          <div
            className=""
          >
            CALLING CODE
          </div>
          <div
            className=""
          />
        </li>
        <li
          className=""
        >
          <div
            className=""
          >
            TIMEZONES
          </div>
          <div
            className=""
          />
        </li>
      </ul>
    </div>
  </div>,
]
`;
